util/zenfs.cc

```diff
@@ -69,21 +70,56 @@ Status zenfs_mount(std::unique_ptr<ZonedBlockDevice> &zbd,
   return s;
 }
 
+// Create or check pre-existing aux directory and fail if it is
+// inaccessible by current user and if it has previous data
+int create_aux_path() {
+  struct dirent *dent;
+  DIR *aux_dir;
+  unsigned int nfiles = 0;
+  int ret = 0;
+
+  errno = 0;
+  ret = mkdir(FLAGS_aux_path.c_str(), S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH);
+  if (ret < 0 && EEXIST != errno) {
+    fprintf(stderr, "ZenFS: Error creating directory %s: %s\n",
+            FLAGS_aux_path.c_str(), strerror(errno));
+    return 1;
+  }
+  // The aux_path is now available, check if it is not empty and
+  // the user has access permission
+
+  errno = 0;
+  aux_dir = opendir(FLAGS_aux_path.c_str());
+  if (errno) {
+    fprintf(stderr, "ZenFS: Error opening directory %s: %s\n",
```

Comment:
You can drop the ZenFS prefix(es), in the error messages, it is not needed. In stead, add that the aux directory could not be opened/is not empty/ does not have the right permissions

Reply:
Ok, done.